### PR TITLE
0.19 upgrade

### PR DESCRIPTION
**Pull request by devnacho**
Hi @basti1302, 

I'm trying to upgrade to 0.19 and did the changes mentioned in the 0.19 Upgrade Guide. This is still Work In Progress.

I had to delete the `toString` method as `Debug.toString` is not available for Packages. 

I've also replaced all `Debug.crash` appearances for a hardcoded`first` and left comments explaining that. It is quite aweful but I see it is the pattern that many libraries did to upgrade to 0.19 and replace `Debug.crash`

Thanks for the library 😄

### CODE REVIEW

File: src/Array/NonEmpty.elm [hunk 1]
```diff
@@ -747,15 +711,18 @@ normalizeSliceIndex index nea =
...
+
     else if index > l then
         l
+
     else
         index
 
 
-{-| Removes the element at the given index.
+{-| (Removes the element at the given index.
```
Comment (basti1302) [inline src/Array/NonEmpty.elm]: Why did you add a `(` here?

File: tests/NonEmptyArrayTest.elm [hunk 2]
```diff
@@ -343,7 +343,7 @@ suite =
                     in
                     Expect.equal "element" (NEA.getSelected nea)
             , fuzz
-                (Fuzz.map2 (,)
+                (Fuzz.map2 (\a b -> ( a, b ))
```
Comment (basti1302) [inline tests/NonEmptyArrayTest.elm]: would id be possible to use `Tuple.pair` here? See https://github.com/elm/compiler/blob/master/upgrade-docs/0.19.md#changes

Or did you intentionally choose the the lambda here because you find it more readable? Just curious.

There are a few places where `(,)` has been replaced like this, so if we end up using something else as a replacement for `(,)`, we should do so consistently.

File: src/Array/NonEmpty.elm [hunk 3]
```diff
@@ -1,38 +1,12 @@
...
-        )
+module Array.NonEmpty exposing
+    ( NonEmptyArray
+    , fromElement, initialize, repeat, fromList, fromArray
+    , length, get, getFirst
+    , set, update, push, append, slice, removeAt, removeAtSafe
+    , selectedIndex, setSelectedIndex, setSelectedIndexAndReport, getSelected, updateSelected, mapSelected, indexedMapSelected
+    , toArray, toList, toIndexedList
+    , foldl, foldr, filter, map, indexedMap
+    )
```
Comment (basti1302) [inline src/Array/NonEmpty.elm:9]: Was the list of exports reordered and formatted by `elm-upgrade`  or a newer version of `elm-format`?

**Review by basti1302 — COMMENTED** [review]
LGTM overall.

I have some minor questions (see inline review comments).

Also:

* Could you also update `README.md` and remove the paragraph starting with "The implementation uses Skinney/elm-array-exploration" and replace it simply with "This package has no dependencies besides elm/core"?
* Travis currently fails, I guess .travis.yml has to be updated as well. Would you like to take a shot at it? Otherwise I can also have a look there.

Thanks for updating this! 🙇  Really appreciated.

File: src/Array/NonEmpty.elm (hunk 3 above)
Comment (devnacho) [reply src/Array/NonEmpty.elm:9]: Hi @basti1302 I used `elm-upgrade` and `elm-format` and it did this strange reorder.

I'm seeing now that if I run manually `elm-format` with `elm-format-0.18 0.5.2-alpha` it changes this and a bunch of other things. (Mostly it indenting stuff)

Should I push the changes with  `elm-format-0.18 0.5.2-alpha` ?

File: tests/NonEmptyArrayTest.elm (hunk 2 above)
Comment (devnacho) [reply tests/NonEmptyArrayTest.elm]: @basti1302 I just run `elm-upgrade` and it did that.

I can definitely change occurrences like this one to `Tuple.pair` if you prefer.

File: src/Array/NonEmpty.elm (hunk 1 above)
Comment (devnacho) [reply src/Array/NonEmpty.elm]: @basti1302 I'll remove it. Not sure if it was `elm-upgrade` or I slipped it unintentionally.

File: tests/NonEmptyArrayTest.elm (hunk 2 above)
Comment (basti1302) [reply tests/NonEmptyArrayTest.elm]: If `elm-upgrade` did this automatically I'm also fine with using leaving the lambda, although I'd find the Tuple.pair style slightly preferable. Not super important either way.

File: src/Array/NonEmpty.elm (hunk 3 above)
Comment (basti1302) [reply src/Array/NonEmpty.elm:9]: No, if a more recent version of elm-format than "elm-format-0.18 0.5.2-alpha" did this, leave it like this.

**Comment by devnacho** [comment]
Hi @basti1302 thanks for your review and your comments 😄 

Regarding tests if you could look into it would be really appreciated as I'm a bit confused.

If you see other packages like:
https://github.com/mdgriffith/elm-style-animation 
or https://github.com/rtfeldman/elm-validate

Both tests have elm 0.18 on `elm-package.json` inside the `tests` folder.

Many other 0.19 projects don't even have tests 🤷‍♂️ 
https://github.com/elm-community/random-extra

**Comment by devnacho** [comment]
Hi @basti1302 how are you? sorry to bother.

Have you seen my previous comment? What do we do with the tests?

**Comment by basti1302** [comment]
I was able to fix the tests for Elm 0.19 today.

This has landed in a44a340a9b8d045623f796b8ba6a4883f08bce79. Version 2.0.0 has been published which includes this and is compatible with Elm 0.19 now.

Stellar work @devnacho! Thanks a lot for taking care of this. 👍

**Comment by devnacho** [comment]
Wohooo! Thanks @basti1302
